server/routes/activities/record-attendance/handlers/activities.ts

```diff
@@ -137,13 +139,17 @@ export default class ActivitiesRoutes {
 
   POST_ATTENDANCES = async (req: Request, res: Response): Promise<void> => {
     const { selectedInstanceIds, activityDate, sessionFilters } = req.body
+    const selectedInstanceIdsArr = selectedInstanceIds ? convertToArray(selectedInstanceIds) : []
     req.session.recordAttendanceRequests = {
       mode: AttendActivityMode.MULTIPLE,
-      selectedInstanceIds: selectedInstanceIds ? convertToArray(req.body.selectedInstanceIds) : [],
+      selectedInstanceIds: selectedInstanceIdsArr,
       activityDate,
       sessionFilters,
     }
-    res.redirect('/activities/attendance/activities/attendance-list')
+    if (selectedInstanceIdsArr.length === 1) {
+      return res.redirect(`/activities/attendance/activities/${selectedInstanceIdsArr[0]}/attendance-list`)
```

Comment:
If only one session is selected then use the "old" attendance list view that can then allow the user to print and cancel a session